node/src/components/consensus/protocols/highway.rs

```diff
@@ -165,16 +165,17 @@ impl<I: NodeIdT, C: Context + 'static> HighwayProtocol<I, C> {
         let max_round_exp = params.max_round_exp();
         let round_exp = params.init_round_exp();
         let start_timestamp = params.start_timestamp();
+        let round_success_meter = prev_cp
+            .and_then(|cp| cp.as_any().downcast_ref::<HighwayProtocol<I, C>>())
```

Comment:
Could this be moved further up, so that we don't downcast twice in the same method?
(And `round_exp` is also redundant: it's the same as `init_round_exp`.)

Reply:
I don't think it's going to be cleaner but if you insist.

Reply:
Actually, it's getting so ugly that I prefer to leave it as it is now.